derp/derphttp/mesh_client.go

```diff
@@ -91,11 +91,11 @@ func (c *Client) RunWatchConnectionLoop(ctx context.Context, ignoreServerKey key
 	}
 
 	sleep := func(d time.Duration) {
-		t := time.NewTimer(d)
+		t, tChannel := c.clock.NewTimer(d)
```

Comment:
I'd need to see more context on this, but it might be possible to replace this with an `Advance`. Right now you're probably going to end up with a test that never finishes.

Reply:
Nevermind, I missed that this _isn't_ in a test.